plugin/call_hierarchy.py

```diff
@@ -0,0 +1,157 @@
+    ) -> List[CallHierarchyItem]:
+        return [incoming_call['from'] for incoming_call in response] if response else []
+
+    def _handle_outgoing_calls_async(
+        self, response: Optional[List[CallHierarchyOutgoingCall]]
+    ) -> List[CallHierarchyItem]:
+        return [outgoing_call['to'] for outgoing_call in response] if response else []
+
+
+class LspCallHierarchyCommand(LspTextCommand):
+
+    capability = 'callHierarchyProvider'
+
+    def is_visible(self, event: Optional[dict] = None, point: Optional[int] = None) -> bool:
+        if self.applies_to_context_menu(event):
+            return self.is_enabled(event, point)
+        return True
+
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None, point: Optional[int] = None) -> None:
+        self._window = self.view.window()
+        session = self.best_session(self.capability)
+        if not session:
+            return
+        position = get_position(self.view, event, point)
+        if position is None:
+            return
+        params = cast(CallHierarchyPrepareParams, text_document_position_params(self.view, position))
+        request = Request.prepareCallHierarchy(params, self.view)
+        session.send_request_async(request, partial(self._handle_response_async, session.config.name))
+
+    def _handle_response_async(self, session_name: str, response: Optional[List[CallHierarchyItem]]) -> None:
+        if not self._window or not self._window.is_valid():
+            return
+        if not response:
+            self._window.status_message("Call hierarchy not available")
+            return
+        data_provider = CallHierarchyDataProvider(
+            self._window, session_name, CallHierarchyDirection.IncomingCalls, response)
+        header = 'Call Hierarchy: Callers of… <a href="{}" title="Show outgoing calls">&#8644;</a>'.format(
+            make_toggle_command(session_name, CallHierarchyDirection.OutgoingCalls, response))
+        new_tree_view_sheet(self._window, "Call Hierarchy", data_provider, header, flags=sublime.ADD_TO_SELECTION)
+
+
+class LspCallHierarchyToggleCommand(LspWindowCommand):
+
+    capability = 'callHierarchyProvider'
+
+    def run(
+        self, session_name: str, direction: CallHierarchyDirection, root_elements: List[CallHierarchyItem]
+    ) -> None:
+        if direction == CallHierarchyDirection.IncomingCalls:
+            current_label = 'Callers of…'
+            new_direction = CallHierarchyDirection.OutgoingCalls
+            tooltip = 'Show Outgoing Calls'
+        elif direction == CallHierarchyDirection.OutgoingCalls:
+            current_label = 'Calls from…'
+            new_direction = CallHierarchyDirection.IncomingCalls
+            tooltip = 'Show Incoming Calls'
+        else:
+            return
+        header = 'Call Hierarchy: {} <a href="{}" title="{}">&#8644;</a>'.format(
+            current_label, make_toggle_command(session_name, new_direction, root_elements), tooltip)
```

Comment:
Would be nice to reuse `make_command_link` here instead of creating new function. I believe it would be enough to extend it with an optional `title` argument?